pkg/query-service/app/logparsingpipeline/preview.go

```diff
@@ -0,0 +1,254 @@
+package logparsingpipeline
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/logstransformprocessor"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor"
+	"go.signoz.io/signoz/pkg/query-service/collectorsimulator"
+	"go.signoz.io/signoz/pkg/query-service/model"
+	"gopkg.in/yaml.v3"
+)
+
+func SimulatePipelinesProcessing(
+	ctx context.Context,
+	pipelines []Pipeline,
+	logs []model.SignozLog,
```

Comment:
Will the input be a structured log or a bunch of log lines in an unstructured string form?